Calculator.yashsachdev/Calculator.cs

```diff
@@ -0,0 +1,121 @@
+﻿namespace Calculator.yashsachdev;
+class Calculator
+{
+
+    private List<double> _storeValue = new List<double>();
+    public List<double> store // allowing access to the field to read-only. 
+    {
+        get { return _storeValue; }
+    }
+
+    private int _count = 0;
+
+    public int cnt
+    {
+        get { return _count; }
+    }
+
+    public double DoOperation(double num1, double num2, string op)
+    {
+        double result = double.NaN;
+
+        switch (op)
+        {
+            case "a":
+                result = num1 + num2;
+                _storeValue.Add(result);
+                break;
+            case "s":
+                result = num1 - num2;
+                _storeValue.Add(result);
+                break;
+            case "m":
+                result = num1 * num2;
+                _storeValue.Add(result);
+                break;
+            case "d":
+
+                if (num2 != 0)
+                {
+                    result = num1 / num2;
+                    _storeValue.Add(result);
+                }
+                break;
+            //challenge 4:Add extra calculations: Square Root, Taking the Power, 10x, Trigonometry functions.
+            case "^":
```

Comment:
This one is a bit personal but I like to keep my switch statements lean. You could refactor this way for example. You can quickly refactor it by selecting a block of code and using "ctrl + ." or right-clicking and choosing quick refactorings. 
![image](https://user-images.githubusercontent.com/108711476/212761924-c6de0af2-4ab7-43df-85fd-4ebe2a3c8be0.png)

![image](https://user-images.githubusercontent.com/108711476/212762063-ab277a5e-8500-495d-9eae-1be75ff748e4.png)